pkg/autodiscovery/autoconfig.go

```diff
@@ -198,34 +201,35 @@ func (ac *AutoConfig) AddConfigProvider(provider providers.ConfigProvider, shoul
 func (ac *AutoConfig) LoadAndRun() {
 	scheduleAll := ac.getAllConfigs()
 	ac.applyChanges(scheduleAll)
-	ac.ranOnce.Store(true)
+
 	log.Debug("LoadAndRun done.")
+
+	ac.m.Lock()
```

Comment:
should the lock be at the start of the function? do we want to protect running several time `ac.applyChanges(scheduleAll)`?